### PR TITLE
Allow users to specify optimizer and learning rate scheduler

### PR DESCRIPTION
**Pull request by bnels**
Addresses #46 

Example from config:
```yaml
training:
  optimizer: Adam
  optimizer_args:
      lr: 0.0025
      weight_decay: 0.005
```
if the user also sets 'learning_rate' then a warning is displayed, and `optimizer_args['lr']` is used.  If a user doesn't specify `optimizer_args` then the old `learning_rate` is used, so changes are backward compatible.

Any valid optimizer in `torch.optim` may be used.

### CODE REVIEW

**Comment by bnels** [comment]
updated yaml, following Kazu's suggestion.  Now everything is in the same yaml block
```yaml
training:
  optimizer:
      name: Adam
      args:
          lr: 0.0025
          weight_decay: 0.005
```

**Comment by bnels** [comment]
I've also added support for a learning rate scheduler, addressing #15 

yaml is similar to setting the optimizer.  Any valid scheduler from `torch.optim.lr_scheduler` (https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate) is supported.

Note that scheduler is stepped every iteration, not every epoch, so the appropriate modifications to parameters should be taken into account.

example config:
```yaml
training:
  seed: 0
  optimizer:
      name: Adam
      args:
          lr: 0.0025
          weight_decay: 0.005
  lr_scheduler:
      name: StepLR
      args:
          step_size: 1000 # adjust every 1000 iterations
...
```
